packages/preview-middleware-client/src/cpe/changes/service.ts

```diff
@@ -310,6 +310,14 @@ export class ChangeService {
                 value,
                 controlName: command.getElement().getMetadata().getName().split('.').pop() ?? ''
             };
+        } else if (changeType === 'addXMLAtExtensionPoint') {
+            result = {
+                type: 'pending',
+                controlId: selectorId,
+                changeType,
+                isActive: index >= inactiveCommandCount,
+                controlName: command.getSelector().name ?? ''
+            };
         } else {
```

Comment:
Here we have duplicated code, so can we remove this else if statement and just in the result controlName put a ternary operator something like this: `result = {
                type: 'pending',
                controlId: selectorId,
                changeType,
                isActive: index >= inactiveCommandCount,
                controlName:
                    changeType === 'addXMLAtExtensionPoint'
                        ? command.getSelector().name ?? ''
                        : command.getElement().getMetadata().getName().split('.').pop() ?? ''
            };`

Reply:
Changed in: [9545333](https://github.com/SAP/open-ux-tools/pull/1527/commits/9545333ef711c253902fbde4a44b999f8790166a)